pkg/client/logs.go

```diff
@@ -110,30 +116,46 @@ func (r *Reader) Read(p []byte) (int, error) {
 	return len(data), nil
 }
 
-// getPodsForLogs retrieves the pods to stream logs from. If a plugin name has been provided, retrieve the pods with
+// getPodsToStreamLogs retrieves the pods to stream logs from. If a plugin name has been provided, retrieve the pods with
 // only the plugin label matching that plugin name. If no pods are found, or no plugin has been specified, retrieve
 // all pods within the namespace.
-func getPodsForLogs(client kubernetes.Interface, cfg *LogConfig) (*v1.PodList, error) {
-	if cfg.Plugin != "" {
-		selector := metav1.AddLabelToSelector(&metav1.LabelSelector{}, "sonobuoy-plugin", cfg.Plugin)
-		options := metav1.ListOptions{LabelSelector: metav1.FormatLabelSelector(selector)}
-		pods, err := client.CoreV1().Pods(cfg.Namespace).List(options)
-		if err != nil {
-			return nil, errors.Wrap(err, "failed to list pods")
-		}
-
-		if len(pods.Items) != 0 {
-			return pods, nil
+func getPodsToStreamLogs(client kubernetes.Interface, cfg *LogConfig, stop chan bool, podListChan chan v1.PodList, errc chan error) {
+	for {
+		select {
+		default:
+			if cfg.Plugin != "" {
```

Comment:
I think an extra layer of abstraction would help clarify the flow here.

The existing `getPodsForLogs` method can remain as-is and you then have a new method which will do polling using that method, log errors, etc.

I think trying to push all the logic into this method makes it much harder to read/understand.